libs/vkd3d/command.c

```diff
@@ -1953,223 +1953,6 @@ static void vk_access_and_stage_flags_from_d3d12_resource_state(const struct d3d
         FIXME("Unhandled resource state %#x.\n", unhandled_state);
 }
 
-static bool vk_barrier_parameters_from_d3d12_resource_state(struct d3d12_device *device, unsigned int state,
-        unsigned int stencil_state, const struct d3d12_resource *resource, VkQueueFlags vk_queue_flags,
-        VkAccessFlags *access_mask, VkPipelineStageFlags *stage_flags, VkImageLayout *image_layout)
-{
-    bool is_swapchain_image = resource && (resource->flags & VKD3D_RESOURCE_PRESENT_STATE_TRANSITION);
-    VkPipelineStageFlags queue_shader_stages = 0;
-
-    if (vk_queue_flags & VK_QUEUE_GRAPHICS_BIT)
-    {
-        queue_shader_stages |= VK_PIPELINE_STAGE_VERTEX_SHADER_BIT
-                | VK_PIPELINE_STAGE_TESSELLATION_CONTROL_SHADER_BIT
-                | VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT
-                | VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT
-                | VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT;
-    }
-    if (vk_queue_flags & VK_QUEUE_COMPUTE_BIT)
-        queue_shader_stages |= VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT;
-
-    switch (state)
-    {
-        case D3D12_RESOURCE_STATE_COMMON: /* D3D12_RESOURCE_STATE_PRESENT */
-            /* The COMMON state is used for ownership transfer between
-             * DIRECT/COMPUTE and COPY queues. Additionally, a texture has to
-             * be in the COMMON state to be accessed by CPU. Moreover,
-             * resources can be implicitly promoted to other states out of the
-             * COMMON state, and the resource state can decay to the COMMON
-             * state when GPU finishes execution of a command list. */
-            if (is_swapchain_image)
```

Comment:
Don't we need to handle swapchain images at least?

Reply:
I'll double-check what wine's DXGI does.

Reply:
Yeah, this is going to need special care; basically we'll have to transition the image layout like we did before and hope that we never receive a swap chain image for anything other than actual presentation. DXGI unfortunately uses D3D12 commands for barriers but raw Vulkan commands for the blit operation.

Reply:
Is this implemented somewhere? It seems to be the only missing piece.

Reply:
Not yet.

Reply:
Ok, I'll do a final review when that is implemented.

Reply:
Done, see last commit of the series. It's a bit dodgy since DXGI makes assumptions about the exact image layout.